google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryWatchTest.java

```diff
@@ -384,6 +384,36 @@ public void limitToLast() throws Exception {
     listenerAssertions.addedIdsIsAnyOf(emptyList(), asList("doc2", "doc3"));
   }
 
+  @Test
+  public void shutdownNowRejectsListener() throws Exception {
+    Query query = randomColl.whereEqualTo("foo", "bar");
+    QuerySnapshotEventListener listener =
+        QuerySnapshotEventListener.builder().setExpectError().build();
+
+    query.addSnapshotListener(listener);
+    firestore.shutdownNow();
+
+    listener.eventsCountDownLatch.awaitError();
+
+    ListenerAssertions listenerAssertions = listener.assertions();
+    listenerAssertions.hasError();
+  }
+
+  @Test
+  public void shutdownNowPreventsListener() throws Exception {
```

Comment:
nit: inclomplete statement for test name. shutdownNow prevents listener from what?

Reply:
Renamed to shutdownNowPreventsAddingNewListener()